File: php/features.php
<?php
declare(strict_types=1);

// Weather SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WeatherFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WeatherBaseFeature();
            case "test":
                return new WeatherTestFeature();
            default:
                return new WeatherBaseFeature();
        }
    }
}

File: py/weather_sdk/features.py
# Weather SDK feature factory

from weather_sdk.feature.base_feature import WeatherBaseFeature
from weather_sdk.feature.test_feature import WeatherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WeatherBaseFeature(),
        "test": lambda: WeatherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
